ERCS/erc-7794.md

```diff
@@ -0,0 +1,820 @@
+   */
+  event StartDateSet(bytes32 indexed grantId, uint256 startDate);
+
+  /**
+   * @dev Emitted when a new milestone date is added to the grant.
+   * @param grantId The unique identifier of the grant.
+   * @param milestoneDate The timestamp of the added milestone.
+   */
+  event MilestoneDateAdded(bytes32 indexed grantId, uint256 milestoneDate);
+
+  /**
+   * @dev Emitted when a milestone date is removed from the grant.
+   * @param grantId The unique identifier of the grant.
+   * @param milestoneDate The timestamp of the removed milestone.
+   */
+  event MilestoneDateRemoved(bytes32 indexed grantId, uint256 milestoneDate);
+
+  /**
+   * @dev Emitted when a disbursement is added to a milestone.
+   * @param grantId The unique identifier of the grant.
+   * @param milestoneDate The timestamp of the milestone.
+   * @param fundingToken The token used for the disbursement.
+   * @param fundingAmount The amount of the disbursement.
+   */
+  event DisbursementAdded(
+    bytes32 indexed grantId,
+    uint256 milestoneDate,
+    address indexed fundingToken,
+    uint256 fundingAmount
+  );
+
+  /**
+   * @dev Emitted when a disbursement is removed from a milestone.
+   * @param grantId The unique identifier of the grant.
+   * @param milestoneDate The timestamp of the milestone.
+   */
+  event DisbursementRemoved(bytes32 indexed grantId, uint256 milestoneDate);
+
+  /**
+   * @dev Emitted when a disbursement status is updated.
+   * @param grantId The unique identifier of the grant.
+   * @param milestoneDate The timestamp of the milestone.
+   * @param isDisbursed Boolean indicating if the disbursement has been made.
+   */
+  event DisbursementMade(
+    bytes32 indexed grantId,
+    uint256 milestoneDate,
+    bool isDisbursed
+  );
+
+  /**
+   * @dev Emitted when an external link is added to the grant.
+   * @param grantId The unique identifier of the grant.
+   * @param link The external URL added.
+   */
+  event ExternalLinkAdded(bytes32 indexed grantId, string link);
+
+  /**
+   * @dev Emitted when an external link is removed from the grant.
+   * @param grantId The unique identifier of the grant.
+   * @param link The external URL removed.
+   */
+  event ExternalLinkRemoved(bytes32 indexed grantId, string link);
+
+  /**
+   * @dev Stores the core details of a grant.
+   * @param id The unique identifier of the grant.
+   * @param chainid The chain ID where the grant is registered.
+   * @param community The name of the community or protocol issuing the grant.
+   */
+  struct Grant {
+    uint256 id;
+    uint256 chainid;
+    string community;
+  }
+
+  /**
+   * @dev Stores the participants related to the grant.
+   * @param grantManager The address of the manager responsible for the grant.
+   * @param grantees A set of addresses representing the grantees involved in the grant.
+   */
+  struct Participants {
+    address grantManager;
+    EnumerableSet.AddressSet grantees;
```

Comment:
This is creating a hard dependency on a particular library, which is arguably an implementation detail. If an implementer wanted to use something other than OZ, this would be a problem. Is there an alternative way to specify this that doesn't rely on a single library?